server/src/language-service/diagnosticsProvider.ts

```diff
@@ -16,9 +16,26 @@ import {
     SepticReferenceProvider,
     AlgParsingError,
     AlgParsingErrorType,
+    SepticComment,
 } from "../septic";
 import { SettingsManager } from "../settings";
 
+export const disableDiagnosticRegex =
+    /\/\/\s+NoQA(\s[\s\w,]+)?|\{#\s+NoQA(\s[\s\w,]+)?\s*#\}/;
```

Comment:
Mixed case is harder to remember and doesn't look very good, imho. Perhaps follow e.g. Python's flake8 nomenclature: 
https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors

If you prefer the capitalization NoQA then at least allow for any case combination (NOQA, NoQA, noqa etc)